lib/dao.js

```diff
@@ -68,8 +67,8 @@ function convertSubsetOfPropertiesByType(inst, data) {
     // Convert the properties by type
     typedData[key] = inst[key];
     if (typeof typedData[key] === 'object' &&
-            typedData[key] !== null &&
-            typeof typedData[key].toObject === 'function') {
```

Comment:
There are a lot of style changes mixed with functional changes. Please either keep the existing style or commit the style changes in a separate one.